### DISCUSSION TITLE
Implement DirectProperty (or Styled) of type ObservableCollection<T> and how to track source changes?

### DISCUSSION THREAD

**Discussion by MihaMarkic** [AvaloniaUI/Avalonia#16715] [Q&A]
Let's say I want to have a property of type ObservableCollection<T>, i.e. a classic implementation
```csharp
public static readonly DirectProperty<SomeControl, ObservableCollection<SomeType>?> MyProperty =
        AvaloniaProperty.RegisterDirect<SomeControl, ObservableCollection<SomeType>?>(nameof(My),
            o => o.My,
            (o, v) => o.My= v,
            defaultBindingMode: BindingMode.OneWay);
ObservableCollection<SomeType>? my;
public ObservableCollection<SomeType>? My
    {
        get => _my;
        set => SetAndRaise(MyProperty, ref _my, value);
    }
```
Now the question is where do I attach to `CollectionChanged` event? And even more important where do I detach from it.
I could attach to it in `My` setter (or better, in `override OnPropertyChanged`) and detach from in `OnDetachedFromLogicalTree` or `OnDetachedFromVisualTree` but I think there might be a bettter way, or?

**Comment by timunie** [comment]
Check out how ItemsControl does handle ItemsSource in source code. You can also inherit this control and have ItemsSource support for free 😊
